File: 01 - Trilha interativa de desafios de programação para iniciantes/Semana 3 - Mini projetos/08/Calculadora_com_Funcoes.py
# Decription: Calculadora com funções
def somar(a, b):
    return a + b

def subtrair(a, b):
    return a - b

def multiplicar(a, b):
    return a * b

def dividir(a, b):
    if b != 0:
        return a / b
    else:
        return "Erro: Divisão por zero"
    
def pedir_numeros():
    num1 = float(input("Digite um numero: "))
    num2 = float(input("Digite outro numero: "))
    return num1, num2
    
def main():
    print ("\n Escolha a operação: \n")
    print ("1. Somar")
    print ("2. Subtrair")
    print ("3. Multiplicar")
    print ("4. Dividir")
    print ("5. Sair")
    escolha = int(input(""))
    
    if escolha == 1:
        num1, num2 = pedir_numeros()
        resultado = somar(num1, num2)
        print(f"A soma de {num1} e {num2} é: {resultado}")
    elif escolha == 2:
        num1, num2 = pedir_numeros()
        resultado = subtrair(num1, num2)
        print(f"A subtração de {num1} e {num2} é: {resultado}")
    elif escolha == 3:
        num1, num2 = pedir_numeros()
        resultado = multiplicar(num1, num2)
        print(f"A multiplicação de {num1} e {num2} é: {resultado}")
    elif escolha == 4:
        num1, num2 = pedir_numeros()
        resultado = dividir(num1, num2)
        print(f"A divisão de {num1} e {num2} é: {resultado}")
    elif escolha == 5:
        print("Saindo...")
    else:
        print("Opção inválida!")


if __name__ == "__main__":
    main()
    

File: 01 - Trilha interativa de desafios de programação para iniciantes/Semana 3 - Mini projetos/08/Calculadora_com_Funcoes.js
// Description: Calculadora com funções
const readline = require('readline');

function somar(a, b) {
    return a + b;
}  
function subtrair(a, b) {
    return a - b;
} 
function multiplicar(a, b) {
    return a * b;
}
function dividir(a, b) {
    if (b === 0) {
        console.log("Erro: Divisão por zero não é permitida.");
        return null;
    }
    return a / b;
}

function pedirNumeros(rl, callback) {
    rl.question("Digite o primeiro numero: ", (num1) => {
        rl.question("Digite o segundo numero: ", (num2) => {
            callback(parseFloat(num1), parseFloat(num2));
        });
    });
}

function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    console.log(" \nEscolha a operação: \n");
    console.log("1. Somar");
    console.log("2. Subtrair");
    console.log("3. Multiplicar");
    console.log("4. Dividir");
    console.log("5. Sair");
    rl.question("", (opcao) => {
        if (opcao === "1") {
            pedirNumeros(rl, (num1, num2) => {
                console.log(`A soma de ${num1} e ${num2} é: ${somar(num1, num2)}`);
                rl.close();
            });
        } else if (opcao === "2") {
            pedirNumeros(rl, (num1, num2) => {
                console.log(`A subtração de ${num1} e ${num2} é: ${subtrair(num1, num2)}`);
                rl.close();
            });
        } else if (opcao === "3") {
            pedirNumeros(rl, (num1, num2) => {
                console.log(`A multiplicação de ${num1} e ${num2} é: ${multiplicar(num1, num2)}`);
                rl.close();
            });
        } else if (opcao === "4") {
            pedirNumeros(rl, (num1, num2) => {
                const resultado = dividir(num1, num2);
                if (resultado !== null) {
                    console.log(`A divisão de ${num1} e ${num2} é: ${resultado}`);
                }
                rl.close();
            });
        } else if (opcao === "5") {
            console.log("Saindo...");
            rl.close();
        } else {
            console.log("Opção inválida. Tente novamente.");
            main();
        }
    });

}

main();